src/coreclr/jit/optimizer.cpp

```diff
@@ -6145,7 +6145,12 @@ bool Compiler::optNarrowTree(GenTree* tree, var_types srct, var_types dstt, Valu
 
                 if (doit && (dstSize <= genTypeSize(tree->gtType)))
                 {
-                    tree->gtType = genSignedType(dstt);
+                    if (!varTypeIsSmall(dstt))
```

Comment:
Does it mean that we will change "UINT" to "INT" but "USHORT" will stay as "USHORT"?

Reply:
Yep 😔. See also https://github.com/dotnet/runtime/pull/52418#issuecomment-841652968.